lib/system/threads.nim

```diff
@@ -47,14 +47,22 @@
 when not declared(ThisIsSystem):
   {.error: "You must not import this module explicitly".}
 
-const
-  StackGuardSize = 4096
-  ThreadStackMask =
-    when defined(genode):
-      1024*64*sizeof(int)-1
-    else:
-      1024*256*sizeof(int)-1
-  ThreadStackSize = ThreadStackMask+1 - StackGuardSize
+when defined(zephyr):
+  const
+    StackGuardSize {.intdefine.} = 128
+    ThreadStackSize {.intdefine.} = 8192
+else:
+  const
+    StackGuardSize = 4096
+    ThreadStackMask =
+      when defined(genode):
+        1024*64*sizeof(int)-1
+      elif defined(zephyr):
+        8192
```

Comment:
I'm not sure I understand the code correctly, but is there any reason other branches (Genode or the default one) have `-1` but Zephyer branch is just 8192 and not 8191?

Reply:
Good question. Also like the logic got mixed up. I'll fix that.

Could the -1 be related to ARC'S ref count to keep the total area on a power of Two?

Reply:
It's a *mask*, a *bitmask*, these work better when every important bit is 1. That said, this bitmask is unused anyway.

Reply:
Ah, yah that makes sense. I was confused because the bitmask isn't used to mask anything.